.github/workflows/release.yml

```diff
@@ -13,7 +14,58 @@ env:
   IMAGE_NAME: ${{ github.repository }}
 
 jobs:
+  test-kubo:
+    name: 'test:kubo'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: L1-node
+
+      - name: Checkout Kubo fork
+        uses: actions/checkout@v3
+        with:
+          repository: filecoin-saturn/kubo
+          ref: refs/heads/saturn
+          path: kubo
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install Kubo gateway test dependencies
+        run: |
+          cd $GITHUB_WORKSPACE/kubo
+          make test_sharness_deps
+
+      - name: Run Kubo gateway tests
+        run: |
+          export L1_NODE_REPO_PATH=$GITHUB_WORKSPACE/L1-node
+          #export L1_NODE_LOG_FILE=/tmp/node.log
+
+          cd $GITHUB_WORKSPACE/kubo/test/sharness
+
+          for file in $(find .  -maxdepth 1 -name 't011*-gateway*.sh' -type f); do
+            echo
+            if [[ $file == *"t0111"* || $file == *"t0114"* ]]; then
+              echo Skip $file
+              continue
+            fi
+            echo Running $file
+            bash $file
+          done
```

Comment:
Do you know if shell scripts here have `-x` or something like that set, so they fail properly?

Reply:
From my testing, the workflow step will fail if any of the tests fail. 
I can make it more explicit with `bash $file || exit 1;`

Reply:
If that's what it does then I think we can rely on the convention without cluttering things up

Reply:
Nvm turns out `bash $file || exit 1;` is needed.